Mathlib/Algebra/TrivSqZeroExt.lean

```diff
@@ -738,6 +738,68 @@ def inlHom [Semiring R] [AddCommMonoid M] [Module R M] [Module Rᵐᵒᵖ M] : R
 
 end Mul
 
+section Inv
+
+variable {R : Type u} {M : Type v}
+
+instance inv [Neg M] [Inv R] [SMul Rᵐᵒᵖ M] [SMul R M] : Inv (tsze R M) :=
+  ⟨fun b => (b.1⁻¹,  - (b.1⁻¹ •> (b.2 <• b.1⁻¹)))⟩
+
+theorem fst_inv [Neg M] [Inv R] [SMul Rᵐᵒᵖ M] [SMul R M]
+    (x : tsze R M) : fst x⁻¹ = (fst x)⁻¹ :=
+  rfl
+
+theorem snd_inv [Neg M] [Inv R] [SMul Rᵐᵒᵖ M] [SMul R M] (x : tsze R M) :
+    snd x⁻¹ = - ((fst x)⁻¹ •> (snd x <• (fst x)⁻¹)) :=
+  rfl
+
+protected theorem mul_inv_cancel [DivisionRing R] [AddCommGroup M]
+    [Module Rᵐᵒᵖ M] [Module R M] {x : tsze R M}
+    (hx : fst x ≠ 0) : x * x⁻¹ = 1 := by
+  ext
+  · rw [fst_mul, fst_inv, fst_one, DivisionRing.mul_inv_cancel (fst x) hx]
+  rw [snd_mul, snd_inv, smul_neg, ← smul_assoc]
+  rw [smul_eq_mul, DivisionRing.mul_inv_cancel (fst x) hx]
+  rw [one_smul, fst_inv, add_left_neg, snd_one]
```

Comment:
When you have adjacent rewrites like this, we would normally combine them. Here's that version, also with some steps merged together by finding slightly more relevant lemmas:
```suggestions
  ext
  · rw [fst_mul, fst_inv, fst_one, mul_inv_cancel hx]
  · rw [snd_mul, snd_inv, snd_one, smul_neg, smul_smul, mul_inv_cancel hx, one_smul, fst_inv,
      add_left_neg]
```
Note also that for symmetry this puts both the fist and second goal in `·`.

Can you make a similar change to the proofs below?